ring.go

```diff
@@ -59,6 +59,20 @@ func NewKeyRingFromPublicKeys(curve types.Curve, pubkeys []types.Point, privkey
 	}, nil
 }
 
+// NewFixedKeyRingFromPublicKeys takes public keys and a curve to create a ring
+func NewFixedKeyRingFromPublicKeys(curve types.Curve, pubkeys []types.Point) (*Ring, error) {
+	size := len(pubkeys)
+	newRing := make([]types.Point, size)
+	for i := 0; i < size; i++ {
+		newRing[i] = pubkeys[i]
```

Comment:
```suggestion
		newRing[i] = pubkeys[i].Copy()
```

probably want to copy the pubkey passed in